pandas/_libs/tslib.pyx

```diff
@@ -5736,3 +5771,22 @@ cdef _calc_julian_from_U_or_W(int year, int week_of_year,
 
 # def _strptime_time(data_string, format="%a %b %d %H:%M:%S %Y"):
 #     return _strptime(data_string, format)[0]
+
+cdef _calc_julian_from_V(int iso_year, int iso_week, int iso_weekday):
+    """Calculate the Julian day based on the ISO 8601 year, week, and weekday.
+    ISO weeks start on Mondays, with week 01 being the week containing 4 Jan.
+    ISO week days range from 1 (Monday) to 7 (Sunday)."""
+
+    cdef:
+        int correction, ordinal
+
+    correction = datetime_date(iso_year, 1, 4).isoweekday() + 3
+    ordinal = (iso_week * 7) + iso_weekday - correction
+    # ordinal may be negative or 0 now, which means the date is in the previous
+    # calendar year
+    if ordinal < 1:
```

Comment:
@pytest.mark.parametrize("s, _format, dt", [
 +        ['2015-1-1', '%G-%V-%u', datetime(2014, 12, 29, 0, 0)],
 +        ['2015-1-4', '%G-%V-%u', datetime(2015, 1, 1, 0, 0)],
 +        ['2015-1-7', '%G-%V-%u', datetime(2015, 1, 4, 0, 0)]
 +    ])----- out of these;

 +        ['2015-1-4', '%G-%V-%u', datetime(2015, 1, 1, 0, 0)],
this one tests this case!